### PR TITLE
Implement REST API integration for v2 orders

### PR DESCRIPTION
**Pull request by msaggiorato**
Rearrange REST API class, and make it work for v2 orders, as seamlessly as possible.

### CODE REVIEW

File: includes/class-wc-amazon-payments-advanced-rest-api-controller.php [hunk 1]
```diff
@@ -549,15 +628,19 @@ protected function get_missing_capture_id_request_error( $request ) {
...
 	 *
-	 * @param WP_Post $order_post Order post object.
+	 * @param int $order_post Order post object.
 	 *
-	 * @return bool True if it's valid request.
+	 * @return WP_Post|WP_Error Post object if it's valid, WP_Error if it's invalid.
 	 */
-	protected function is_valid_order( $order_post ) {
+	protected function is_valid_order( $order_id ) {
+		$order_post = get_post( (int) $order_id );
```
Comment (lauramelos) [inline includes/class-wc-amazon-payments-advanced-rest-api-controller.php:636]: why not use the WC_Order object to do the verifications in this method?

File: includes/class-wc-gateway-amazon-payments-advanced.php [hunk 2]
```diff
@@ -1684,4 +1692,109 @@ public function maybe_hide_amazon_buttons() {
 		<?php
 	}
 
+	public function perform_authorization( $order, $capture_now = true, $id = null ) {
+		$order_id = $order->get_id();
```
Comment (lauramelos) [inline includes/class-wc-gateway-amazon-payments-advanced.php:1696]: Should check if the order is an actual WC_Order before access their methods

File: includes/class-wc-gateway-amazon-payments-advanced.php [hunk 3]
```diff
@@ -1684,4 +1692,109 @@ public function maybe_hide_amazon_buttons() {
...
+		}
+
+		wc_apa()->get_gateway()->log_charge_permission_status_change( $order );
+		wc_apa()->get_gateway()->log_charge_status_change( $order, $charge );
+
+		return $charge;
+	}
+
+	public function perform_cancel_auth( $order, $id = null ) {
+		$order_id = $order->get_id();
```
Comment (lauramelos) [inline includes/class-wc-gateway-amazon-payments-advanced.php:1736]: same above

File: includes/class-wc-gateway-amazon-payments-advanced.php [hunk 4]
```diff
@@ -1684,4 +1692,109 @@ public function maybe_hide_amazon_buttons() {
...
+		}
+
+		wc_apa()->get_gateway()->log_charge_permission_status_change( $order );
+		wc_apa()->get_gateway()->log_charge_status_change( $order, $charge );
+
+		return $charge;
+	}
+
+	public function perform_capture( $order, $id = null ) {
+		$order_id = $order->get_id();
```
Comment (lauramelos) [inline includes/class-wc-gateway-amazon-payments-advanced.php:1758]: same above

File: includes/class-wc-amazon-payments-advanced-rest-api-controller.php (hunk 1 above)
Comment (msaggiorato) [reply includes/class-wc-amazon-payments-advanced-rest-api-controller.php:636]: Keeping the same implementation as v1 for backwards compatibility

**Pull request merged**